src/message/fetchActions.js

```diff
@@ -192,11 +192,13 @@ const isFetchNeededAtAnchor = (state: GlobalState, narrow: Narrow, anchor: numbe
 export const fetchMessagesInNarrow = (
   narrow: Narrow,
   anchor: number = FIRST_UNREAD_ANCHOR,
-) => async (dispatch: Dispatch, getState: GetState) => {
+) => async (dispatch: Dispatch, getState: GetState): Promise<Message[] | void> => {
   if (!isFetchNeededAtAnchor(getState(), narrow, anchor)) {
-    return;
+    // Just `return;` would be fine, but ESLint doesn't want us to do
+    // that while elsewhere returning something else.
+    return undefined;
```

Comment:
I think this doesn't even need a comment. Just `return;` would result in the same behavior, but `return undefined;` is a way of being explicit that this codepath is choosing to return `undefined`, in a function that otherwise can return a more interesting value.

(Perhaps a shorter way to say this is: I agree with that ESLint rule :wink: )